commonui/src/main/java/com/hyperwallet/android/ui/common/insight/HyperwalletInsight.java

```diff
@@ -222,11 +232,12 @@ public void trackError(@NonNull final Context context, @NonNull final String pag
             @NonNull final String pageGroup, @NonNull final Map<String, String> errorInfoMap) {
 
         final ErrorInfo errorInfo = new ErrorInfo.ErrorInfoBuilder()
-                .type(errorInfoMap.get(InsightEventTag.InsightEventTagEventParams.ERROR_TYPE))
-                .message(errorInfoMap.get(InsightEventTag.InsightEventTagEventParams.ERROR_MESSAGE))
-                .code(errorInfoMap.get(InsightEventTag.InsightEventTagEventParams.ERROR_CODE))
-                .field(errorInfoMap.get(InsightEventTag.InsightEventTagEventParams.ERROR_FIELD_NAME))
-                .description(errorInfoMap.get(InsightEventTag.InsightEventTagEventParams.ERROR_DESCRIPTION))
+                .type(errorInfoMap.remove(InsightEventTag.InsightEventTagEventParams.ERROR_TYPE))
+                .message(errorInfoMap.remove(InsightEventTag.InsightEventTagEventParams.ERROR_MESSAGE))
+                .code(errorInfoMap.remove(InsightEventTag.InsightEventTagEventParams.ERROR_CODE))
+                .field(errorInfoMap.remove(InsightEventTag.InsightEventTagEventParams.ERROR_FIELD_NAME))
+                .description(errorInfoMap.remove(InsightEventTag.InsightEventTagEventParams.ERROR_DESCRIPTION))
+                .params(errorInfoMap)
```

Comment:
I'm not a fan of this.  Personally, I think we should get rid of ErrorInfo object in favour of a Map so we can take advantage of loose typing. Since in end, they all just get thrown into "event_params"